### DISCUSSION TITLE
Running portable Trillium stored on a network drive

### DISCUSSION THREAD

**Discussion by VladimirKirichenko** [TriliumNext/Trilium#2940] [Q&A] [answered]
Hi all.

I'm using Trillium that is stored on a memory stick. Running Trilium in a portable configuration (via trilium-portable.bat) makes this very easy and efficient.
To improve my workflow even further (i.e. to make Trilium accessible from any computer connected to my network), I've performed following steps:
1) Plugged in my memory stick into my router to make it accessible via Wi-Fi (SMB protocol).
2) Mapped the memory stick as a network drive.

When launching Trillium using trilium-portable.bat, the script seems to work just fine (I get no errors from the script) and after that Trillium window shows up for a split second and crashes.

Does anyone know how to fix this?

UPD: only one user is going to be using Trilium at a time, so no support for multiple users is required.

**Comment by sigaloid** [comment]
Can you open a command prompt in the folder with the bat file, then run the bat file by entering the name of it, so the error can be seen before it closes?

**Reply by VladimirKirichenko** [reply]
There is no error in the command prompt when I run trilium-portable.bat.
Below is everything that gets output to the command prompt window:

```
Microsoft Windows [Version 10.0.19044.1706]
(c) Microsoft Corporation. All rights reserved.
C:\Users\Asus-pc>s:
S:\>cd S:\trilium-windows-x64
S:\trilium-windows-x64>trilium-portable.bat
S:\trilium-windows-x64>SET DIR=S:\trilium-windows-x64\
S:\trilium-windows-x64>SET TRILIUM_DATA_DIR=S:\trilium-windows-x64\\trilium-data
S:\trilium-windows-x64>cd S:\trilium-windows-x64\
S:\trilium-windows-x64>WHERE powershell.exe
C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe
S:\trilium-windows-x64>IF 0 NEQ 0 (start trilium.exe )  ELSE (powershell.exe ./trilium-portable.ps1 )
Security warning
Run only scripts that you trust. While scripts from the internet can be useful, this script can potentially harm your
computer. If you trust this script, use the Unblock-File cmdlet to allow the script to run without this warning
message. Do you want to run S:\trilium-windows-x64\trilium-portable.ps1?
[D] Do not run  [R] Run once  [S] Suspend  [?] Help (default is "D"): R
S:\trilium-windows-x64>
```

At first, I had an issue with PowerShell execution policies, which prohibited execution of any PS scripts, but I've fixed that one by setting execution policy to Unrestricted.

I can see that the script attempts to start trilium.exe because I get another security warning (Windows starts to complain because it can't verify who created executable). However, after I click Run, trilium.exe crashes.

**Comment by zadam** [comment, marked as answer]
I don't know about this particular problem, but generally trying to use Trilium (or any SQLite database) over network drives will sooner or later lead to data corruption. Network drives implementations usually don't have the same guarantees as local FS and/or are buggy.

**Comment by trailjeep** [comment]
I've been bitten by this before also.
@zadam 's advice is best.

Issue: SQLite  Database Always Locked On CIFS/NFS Mount
Reference: [How to make SQLite run safely on CIFS mounted file](https://sqlite-users.sqlite.narkive.com/PkF3Wqkw/how-to-make-sqlite-run-safely-on-cifs-mounted-file-system)
#sqlite #issues #database #lock #cifs #nfs

SQLite3 uses filesystem locking to prevent DB corruption.  CIFS & NFS do not support this so we need to use the following mount options:

Note: Neither of these mount options is ideal, but I needed them for my Raspi server using external storage

- CIFS
  - `nobrl`: Do not send byte range lock requests to the server. This is necessary for certain applications that break with cifs style mandatory byte range locks (and most cifs servers do not yet support requesting advisory byte range locks).
- NFS
  - `nolock`: Selects whether to use the NLM sideband protocol to lock files on the server. If neither option is specified (or if lock is specified), NLM locking is used for this mount point. When using the nolock option, applications can lock files, but such locks provide exclusion only against other applications running on the same client. Remote applications are not affected by these locks.
